lib/taskana-core-test/src/test/java/acceptance/task/create/CreateTaskAccTest.java

```diff
@@ -0,0 +1,967 @@
+    defaultAttachment.setObjectReference(objectReference);
+    testCreateTask.accept(defaultAttachment);
+
+    ObjectReference objectReference1 =
+        ObjectReferenceBuilder.newObjectReference()
+            .system("SYSTEM_B")
+            .systemInstance("INSTANCE_B")
+            .company("COMPANY_A")
+            .type(null)
+            .value("12345678901234567890123456789012345678901234567890")
+            .build();
+    defaultAttachment.setObjectReference(objectReference1);
+    testCreateTask.accept(defaultAttachment);
+
+    ObjectReference objectReference2 =
+        ObjectReferenceBuilder.newObjectReference()
+            .system("SYSTEM_B")
+            .systemInstance("INSTANCE_B")
+            .company(null)
+            .type("ArchiveId")
+            .value("12345678901234567890123456789012345678901234567890")
+            .build();
+    defaultAttachment.setObjectReference(objectReference2);
+    testCreateTask.accept(defaultAttachment);
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_UseCustomName_For_NewTask() throws Exception {
+    Task task = createDefaultTask();
+    task.setName("Test Name");
+
+    Task createdTask = taskService.createTask(task);
+
+    assertThat(createdTask).isNotNull();
+    assertThat(createdTask.getCreator()).isEqualTo("user-1-2");
+    assertThat(createdTask.getName()).isEqualTo("Test Name");
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_UseClassificationMetadataFromCorrectDomain_For_NewTask() throws Exception {
+    ClassificationSummary classificationSummary =
+        ClassificationBuilder.newClassification()
+            .key(defaultClassificationSummary.getKey())
+            .priority(99)
+            .domain("DOMAIN_B")
+            .buildAndStoreAsSummary(classificationService, "admin");
+    WorkbasketSummary workbasketSummary =
+        WorkbasketBuilder.newWorkbasket()
+            .key(defaultWorkbasketSummary.getKey())
+            .domain("DOMAIN_B")
+            .name("Megabasket")
+            .type(WorkbasketType.GROUP)
+            .orgLevel1("company")
+            .buildAndStoreAsSummary(workbasketService, "admin");
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(workbasketSummary.getId())
+        .accessId("user-1-2")
+        .permission(WorkbasketPermission.OPEN)
+        .permission(WorkbasketPermission.READ)
+        .permission(WorkbasketPermission.APPEND)
+        .buildAndStore(workbasketService, "admin");
+    Task task = taskService.newTask(defaultWorkbasketSummary.getKey(), "DOMAIN_B");
+    task.setClassificationKey(classificationSummary.getKey());
+    task.setPrimaryObjRef(defaultObjectReference);
+    task.setOwner("user-1-2");
+    task.setName("Test Name");
+
+    Task createdTask = taskService.createTask(task);
+
+    assertThat(createdTask).isNotNull();
+    assertThat(createdTask.getCreator()).isEqualTo("user-1-2");
+    assertThat(createdTask.getPriority()).isEqualTo(99);
```

Comment:
Please check the classificationSummary and its domain insteadOf checking priority